### PR TITLE
[TASK] Remove version restriction for TYPO3 core

### PR DESCRIPTION
**Pull request by jousch**
With this it' easier to test extension with TYPO3 8 dev-master
in an TYPO3 composer instance.

Resolves: https://forge.typo3.org/issues/77940

### CODE REVIEW

**Comment by IchHabRecht** [comment]
Please do not support unexisting versions. You don't know which changes are coming with 8.4 and so on.

**Comment by jousch** [comment]
Yeah but what is the best way to test the ext in actual projects with the TYPO3 master? It's annoying to have this version constraints. When releasing ext, it's ok to fix to a known working version. Any ideas?

**Comment by IchHabRecht** [comment]
You can install the extension with the EM and force the installation (in non-composer mode) or lower the TYPO3 version with "typo3/cms": "dev-master as 8.2.0" in composer mode

**Comment by Gernott** [comment]
Idea: Due composer problems, we could disable the version check as requested (e.g. set to TYPO3 99.99.99) and let the real check in ext_emconf (TYPO3 8.3.99).
So the composer users are happy and the backend-users are going a secure way.

**Comment by jousch** [comment]
Just set to 8.99.99. and when creating a stable release, the composer.json version constraints should be also fixed for that release.

Maybe there are some better workflows, but that could work.

**Comment by IchHabRecht** [comment]
To be honest I don't get the problem with setting the right TYPO3 dependency. Neither with nor without composer. If there is any breaking change in the core which needs an update of the extension, this very version still claims to be compatible but isn't.

But this isn't my decision. I can just recommend.

**Comment by Gernott** [comment]
Let us discuss this on the TUGA next week. Perhaps Georg Ringer is also there, because he also wrote a ticket on forge.

**Comment by helhum** [comment]
I'm strictly against loosening version constraints of extensions.
It helps now, but creates issues later.

Let's assume TYPO3 8.4 or 8.5 introduces a breaking change, which also breaks mask.
Then it would be possible to install an incompatible mask version with a TYPO3 version 8.5
And there would be **NO** way to communicate, that these versions are not compatible without a [flux capacitor](http://backtothefuture.wikia.com/wiki/Flux_capacitor).

The whole purpose and benefits of composer dependency management would become void.

I learned this the hard way with typo3_console and would never specify unbound (or less strictly bound) version constraints again.

If you want to test mask with current TYPO3 master, you would need to specify a version alias for TYPO3 anyway, so why not specifying `"typo3/cms": "^8.x@dev as 8.2.0"` then to test mask?

And with every TYPO3 minor version release, the version constraints need to be raised in mask (after ensuring compatiblity of course)

In this case, mask should be testet with TYPO3 8.3 and then a bugfix version of mask should be released with adapted version constraints, if mask is compatible.

File: composer.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
 	"license": ["GPL-2.0+"],
 	"version": "2.1.0",
 	"require": {
-		"typo3/cms-core": ">=7.6.0 <8.2.99"
+		"typo3/cms-core": ">=7.6.0 <8.99.99"
```
Comment (helhum) [inline composer.json]: I'd go for `">=7.6.0 <8.4.99"` or even better (readable) `"~7.6.0 || ~8.3.0"`

**Comment by IchHabRecht** [comment]
Not sure if there was any decision yet, but I would recommend to merge #47

**Comment by butu** [comment]
Adjusted to the correct version, that is supported.

**Pull request merged**